src/averageAges.js

```diff
@@ -37,7 +50,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+
+  const motherNames = people.map(person => person.mother);
+
+  const mothers = women.filter(person => motherNames.includes(person.name));
+
+  if (withChildren) {
```

Comment:
avoid `if()` statements in this task. 
![image](https://user-images.githubusercontent.com/70009205/200522970-4d661284-6b79-4d08-bc9f-11178043edd7.png)